joomla/joomla-cms
# Add Development Container Configuration and GitHub Codespace Support

Pull Request for Issue #43802.

### Summary of Changes

This pull request introduces a development environment configuration for Joomla! CMS using devcontainers, making it possible to develop and test Joomla directly from GitHub Codespaces or Visual Studio Code. The implementation is based on existing devcontainer examples, which have been adjusted for Joomla.

Key changes include:

- Addition of a `.devcontainer` folder with all necessary configuration files to initialize a development environment.
- Support for running Joomla CMS seamlessly in GitHub Codespaces or locally in Visual Studio Code.

This configuration simplifies the setup for developers looking to contribute to Joomla's core, providing a consistent and efficient environment with minimal effort.

### Testing Instructions

1. Open the Joomla repository in GitHub Codespaces or Visual Studio Code.
2. Ensure you have the latest version of Visual Studio Code installed, along with the Remote - Containers extension.
3. Launch the devcontainer environment.
4. Verify that Joomla runs correctly by following the on-screen setup instructions.
5. Confirm that changes to the codebase are reflected when testing locally.

### Actual result BEFORE applying this Pull Request

No integrated setup available for Joomla development in GitHub Codespaces or Visual Studio Code using devcontainers.

### Expected result AFTER applying this Pull Request

Developers will be able to use GitHub Codespaces or Visual Studio Code to set up a full Joomla development environment with devcontainers, simplifying contributions and testing.

---
I've tested the implementation, and it works seamlessly in GitHub Codespaces. However, this pull request is intended to start the conversation around configuring additional tools that will ensure proper development practices for Joomla’s core, such as linters, cs fixers, and extensions to adhere to Joomla's standards. Collaboration from the community is highly encouraged to refine this setup and make it more robust for all contributors.
